### PR TITLE
Extract Groups management from HomeDBManager

### PR DESCRIPTION
**Pull request by fflorent**
## Context

I pave the way to implement Groups for SCIM.

## Proposed solution

Refactoring for moving the Groups Management outside of HomeDBManager.

## Related issues

#870 

## Has this been tested?

- [ ] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [x] 🙅 no, because this is not relevant here (no methods are used elsewhere than in `lib/homedb`)
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: app/gen-server/lib/homedb/GroupsManager.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
+import { Group } from "app/gen-server/entity/Group";
+import { GroupDescriptor, NonGuestGroup, Resource } from "app/gen-server/lib/homedb/Interfaces";
+import * as roles from 'app/common/roles';
+import { Permissions } from 'app/gen-server/lib/Permissions';
+import { User } from "app/gen-server/entity/User";
+import { Organization } from "app/gen-server/entity/Organization";
+import { Workspace } from "app/gen-server/entity/Workspace";
+import { Document } from "app/gen-server/entity/Document";
+import { AclRule } from "app/gen-server/entity/AclRule";
```
Comment (paulfitz) [inline app/gen-server/lib/homedb/GroupsManager.ts:9]: Ideally imports are ordered alphabetically, and use of quotes are consistent. Sorry we don't have automatic linting for this.

**Review by paulfitz — COMMENTED** [review]
Looks like a straightforward refactor. One tiny comment, then I think it is good.

File: app/gen-server/lib/homedb/GroupsManager.ts (hunk 1 above)
Comment (fflorent) [reply app/gen-server/lib/homedb/GroupsManager.ts]: @paulfitz I am never really sure about how to sort the imports. How should we sort this line: 
`import { GroupDescriptor, NonGuestGroup, Resource } from "app/gen-server/lib/homedb/Interfaces";` 

Should we consider the imported object (`GroupDescriptor`) or the path (`"app/gen-server/lib/homedb/Interfaces"`) for sorting?

Comment (paulfitz) [reply app/gen-server/lib/homedb/GroupsManager.ts]: The path

**Review by paulfitz — APPROVED** [review]

**Pull request merged**